WcaOnRails/app/models/competition_event.rb

```diff
@@ -21,4 +21,31 @@ def to_wcif
       "rounds" => self.rounds.map(&:to_wcif),
     }
   end
+
+  def load_wcif!(wcif)
+    # Remove extra rounds.
+    self.rounds.each do |round|
+      wcif_round = wcif["rounds"][round.number - 1]
```

Comment:
I hadn't thought about how the document is structured before. I now realize that `Event` contains an array of `Round`, each containing an array of "RoundResult".

That's a massively nested data structure, and far from easy human reading. I'm also a little bothered by mixing "static" data defining the competition format, with "dynamic" results. Maybe I'm just trying to turn JSON into SQL, I don't have any experience with how JSON formats are usually done.

It seems "cleaner" in some sense to have these in 3 separate sections, that refer to each other by keys. That could also indirectly solve the array problem.
---
Not a fan of `.index`-s.

Leaving the WCIF format as is is not the worst thing.